internal/connector/internal/services/connector_cluster.go

```diff
@@ -840,3 +845,15 @@ func (k *connectorClusterService) ResetServiceAccount(ctx context.Context, clust
 	}
 	return nil
 }
+
+func (k *connectorClusterService) ListProcessorDeployments(ctx context.Context, clusterId string, filterChannelUpdates bool, filterOperatorUpdates bool, includeDanglingDeploymentsOnly bool, listArgs *services.ListArguments, gtVersion int64) (dbapi.ProcessorDeploymentList, *api.PagingMeta, *errors.ServiceError) {
```

Comment:
I delegate `processor deployment` cluster-level functions to the `processor deployment` service.

There was a [comment](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/services/connector_deployment.go#L10) for `connector deployment`'s suggesting the existing implementation was to be refactored.

I _think_ I've made an improvement for `processor deployment`'s.